src/compaction/mod.rs

```diff
@@ -180,15 +186,17 @@ where
         delete_gens: &mut Vec<FileId>,
     ) -> Result<(), CompactionError<R>> {
         let mut level = 0;
-
+        let mut compact_status = HashSet::new();
         while level < MAX_LEVEL - 2 {
             if !option.is_threshold_exceeded_major(version, level) {
                 break;
             }
-            let (meet_scopes_l, start_l, end_l) = Self::this_level_scopes(version, min, max, level);
-            let (meet_scopes_ll, start_ll, end_ll) =
+            let meet_scopes_l = Self::this_level_scopes(version, min, max, level);
+            let mut meet_scopes_ll =
                 Self::next_level_scopes(version, &mut min, &mut max, level, &meet_scopes_l)?;
 
+            meet_scopes_ll.retain(|scope| compact_status.insert(scope.gen));
```

Comment:
Is this retain `meet_scopes_l`? Because there will be no new sstable in `meet_scopes_ll` because compaction will increase step by step at level.

Reply:
It should be `meet_scopes_ll`. Because we only care sstables that may be compacted in next compaction loop

Reply:
The `meet_scopes_ll` of this compaction is the `meet_scopes_l` of the next compaction, right?

Reply:
Yes, `meet_scopes_l` also need filtering, but it needn't insert to `compact_status`. So the code may be like this:
```rust
meet_scopes_l.retain(|scope| !compact_status.contains(&scope.gen));
meet_scopes_ll.iter().for_each(|scope| {
    compact_status.insert(scope.gen);
});
```

Reply:
I'm not sure if `compact_status` requires both `meet_scopes_ll` to be inserted, new sstables are ignored here, and is it better to just insert new sstables?